Gemfile

```diff
@@ -1,7 +1,9 @@
 source 'https://rubygems.org'
 gemspec
+
+# Other dependencies are defined in the Appraisals file
 group :development do
   gem 'appraisal', '~> 2.4'
   gem 'rspec'
-  gem 'sqlite3'
+  gem 'sqlite3', '~> 1.4'
```

Comment:
this is the highest version active record 7.0 supports:

https://github.com/rails/rails/blob/v7.0.8.1/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L13